go_modules/spec/dependabot/go_modules/update_checker/latest_version_finder_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require "dependabot/dependency"
+require "dependabot/dependency_file"
+require "dependabot/go_modules/native_helpers"
+require "dependabot/go_modules/update_checker/latest_version_finder"
+
+RSpec.describe Dependabot::GoModules::UpdateChecker::LatestVersionFinder do
+  let(:dependency_name) { "github.com/dependabot-fixtures/go-modules-lib" }
+
+  let(:dependency_version) { "1.0.0" }
+
+  let(:dependency) do
+    Dependabot::Dependency.new(
+      name: dependency_name,
+      version: dependency_version,
+      package_manager: "go_modules",
+      requirements: [{
+        file: "go.mod",
+        requirement: dependency_version,
+        groups: [],
+        source: { type: "default", source: dependency_name }
+      }]
+    )
+  end
+
+  let(:go_mod_content) do
+    <<~GOMOD
+      module foobar
+      require #{dependency_name} v#{dependency_version}
+    GOMOD
+  end
+
+  let(:dependency_files) do
+    [
+      Dependabot::DependencyFile.new(
+        name: "go.mod",
+        content: go_mod_content
+      )
+    ]
+  end
+
+  describe "#latest_version" do
+    context "when the latest version is an '+incompatible' version" do # https://golang.org/ref/mod#incompatible-versions
+      let(:dependency_name) { "github.com/go-yaml/yaml" }
```

Comment:
@mctofu: Do you think this is a situation where we should create our own fixture in the @dependabot-fixtures org?

Reply:
That would guard against any future changes to `github.com/go-yaml/yaml` from breaking the test although that seems less likely in this case. I think we'd just need to setup a go library repo that didn't have a `go.mod` but did have a version tag to reproduce.

Reply:
Thanks, @mctofu. I'll work on getting a fixture for this in @dependabot-fixtures.